Cython/Compiler/AutoDocTransforms.py

```diff
@@ -66,28 +66,47 @@ def _fmt_annotation(self, node):
         # print(type(node).__name__, '-->', result)
         return result
 
-    def _fmt_arg(self, arg):
-        if arg.type is PyrexTypes.py_object_type or arg.is_self_arg:
-            doc = arg.name
-        else:
-            doc = arg.type.declaration_code(arg.name, for_display=1)
+    def _setup_format(self):
+        format_ = self.current_directives['embedsignature.format']
+        self.is_cython = format_ == 'cython'
+        self.is_python = format_ == 'python'
+        self.is_clinic = format_ == 'clinic'
```

Comment:
I think we can afford calling these `format_is_clinic` etc. `self.is_clinic` reads funny and `self.is_python/cython` is also not very clear.

I'm also not a big fan of `underscore_` names. Why not call it `signature_format` or `format_name` instead?